scripts/histogram_test.jl

```diff
@@ -0,0 +1,150 @@
+using GeodesicRendering
+using AccretionGeometry
+using CarterBoyerLindquist
+using AccretionFormulae
+using StaticArrays
+using Printf
+
+using Plots
+gr()
+
+function temperature(m, sol, max_time; kwargs...)
+    g =  AccretionFormulae.redshift(m, sol, max_time; kwargs...)
+
+    u = sol.u[end]
+    M = m.M
+    a_star = m.a
+    temperature = AccretionFormulae.observed_temperature(u[2], a_star, M, g)
+    
+end
+
+function radius(m, sol, max_time; kwargs...)
+    u = sol.u[end]
+    radius = u[2]
+end
+
+function temperature_render(;mass=1, spin=0.998, obs_angle=85.0, disc_angle=90.0, 
+                            tolerance=1e-8, dtmax=1000.0, 
+                            size_multiplier::Int64=1, resolution=400, fov=3.0,
+                            η=0.1, η_phys=0.1, edd_ratio=0.1, edd_ratio_phys=0.1
+                            )
+                            
+    m = CarterMethodBL(M=1.0, a=spin)
+    M = m.M
+
+    # observer position
+    u = [0.0, 1000.0, deg2rad(obs_angle), 0.0]
+    R_isco = AccretionFormulae.r_isco(m.a, m.M)
+
+    # disc
+    d = GeometricThinDisc(R_isco+1, 50.0, deg2rad(disc_angle))
+
+
+    # create and compose the ValueFunctions
+    temperature_vf = (
+        ValueFunction(temperature)
+        ∘ FilterValueFunction((m, sol, max_time; kwargs...) -> sol.u[end][2] > R_isco, NaN)
+        ∘ ConstValueFunctions.filter_early_term
+    )
+
+    radius_vf = (
+        ValueFunction(radius)
+        ∘ FilterValueFunction((m, sol, max_time; kwargs...) -> sol.u[end][2] > R_isco, NaN)
+        ∘ ConstValueFunctions.filter_early_term
+    )
+
+    redshift_vf = (
+        ValueFunction(AccretionFormulae.redshift)
+        ∘ FilterValueFunction((m, sol, max_time; kwargs...) -> sol.u[end][2] > R_isco, NaN)
+        ∘ ConstValueFunctions.filter_early_term   
+    )
+
+    # do the render
+    temperature_img = @time rendergeodesics(
+        m, u, 2000.0, 
+        d, 
+        fov_factor=fov*size_multiplier, abstol=tolerance, reltol=tolerance,
+        image_width = 350*size_multiplier,
+        image_height = 250*size_multiplier,
+        vf = temperature_vf,
+        dtmax = dtmax
+    )
+
+    radius_img = @time rendergeodesics(
+        m, u, 2000.0, 
+        d, 
+        fov_factor=fov*size_multiplier, abstol=tolerance, reltol=tolerance,
+        image_width = 350*size_multiplier,
+        image_height = 250*size_multiplier,
+        vf = radius_vf,
+        dtmax = dtmax
+    )
+
+    redshift_img = @time rendergeodesics(
+        m, u, 2000.0, 
+        d, 
+        fov_factor=fov*size_multiplier, abstol=tolerance, reltol=tolerance,
+        image_width = 350*size_multiplier,
+        image_height = 250*size_multiplier,
+        vf = redshift_vf,
+        dtmax = dtmax
+    )
```

Comment:
Ok, we'll leave this tripple render for now. Should hopefully be able to push the render cache later today, so can tidy this up later.

Reply:
Sounds good